src/org/labkey/test/pages/dataset/EditDatasetDefinitionPage.java

```diff
@@ -0,0 +1,282 @@
+        elementCache().nameInput.set(name);
+        return this;
+    }
+
+    public String getName()
+    {
+        return elementCache().nameInput.get();
+    }
+
+    public EditDatasetDefinitionPage setDescription(String description)
+    {
+        expandPropertiesPanel();
+        elementCache().descriptionInput.set(description);
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return elementCache().descriptionInput.get();
+    }
+
+    // properties panel interactions
+    // get/set category
+    public EditDatasetDefinitionPage setCategory(String category)
+    {
+        expandPropertiesPanel();
+        elementCache().categorySelect.filterSelect(category);
+        return this;
+    }
+
+    public String getCategory()
+    {
+        return elementCache().categorySelect.getValue();
+    }
+
+    // get/set label
+    public EditDatasetDefinitionPage setDatasetLabel(String label)
+    {
+        expandPropertiesPanel();
+        elementCache().labelInput.set(label);
+        return this;
+    }
+
+    public String getDatasetLabel()
+    {
+        return elementCache().labelInput.get();
+    }
+
+    // get/select radio button for: participants only(demographic data)/participants and visit/participants, visit, and additional key field
+    public EditDatasetDefinitionPage setIsDemographicData(boolean checked)
+    {
+        expandPropertiesPanel();
+        elementCache().participantsOnlyRadioBtn.set(checked);
+        return this;
+    }
+
+    public boolean isDemographicsData()
+    {
+        return  elementCache().participantsOnlyRadioBtn.get();
+    }
+
+    public String getAdditionalKeyColDataField()
+    {
+        if (!elementCache().keyFieldSelect.isEnabled())
+            return null;
+        else
+            return elementCache().keyFieldSelect.getValue();
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColDataField(String field)
+    {
+        setAdditionalKeyColumnType(LookupAdditionalKeyColType.MANAGEDFIELD);
+        elementCache().keyFieldSelect.select(field);
+        return this;
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColManagedField(String field)
+    {
+        setAdditionalKeyColumnType(LookupAdditionalKeyColType.MANAGEDFIELD);
+        elementCache().keyFieldSelect.select(field);
+        elementCache().keyPropertyManagedBox.check();
+        return this;
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColumnType(LookupAdditionalKeyColType type)
+    {
+        expandPropertiesPanel();
+        new RadioButton(elementCache().dataRowRadioBtn(type.getIndex()).findElement(getDriver())).check();
+        return this;
+    }
+
+    public boolean isAdditionalKeyManagedEnabled()
+    {
+        expandPropertiesPanel();
+        return elementCache().additionalKeyFieldRadioBtn.isChecked() &&
+                elementCache().keyFieldSelect.hasValue() &&
+                elementCache().keyPropertyManagedBox.isChecked();
```

Comment:
the method name indicates that this is a check for if the checkbox is enabled or not but this last part of the statement would only return true if it was also checked